yt_dlp/extractor/rai.py

```diff
@@ -592,57 +583,62 @@ class RaiIE(RaiBaseIE):
     }, {
-        # Direct MMS URL
+        # Direct MMS: Media URL no longer works.
         'url': 'http://www.rai.it/dl/RaiTV/programmi/media/ContentItem-b63a4089-ac28-48cf-bca5-9f5b5bc46df5.html',
         'only_matching': True,
     }]
 
     def _extract_from_content_id(self, content_id, url):
         media = self._download_json(
             f'https://www.rai.tv/dl/RaiTV/programmi/media/ContentItem-{content_id}.html?json',
-            content_id, 'Downloading video JSON')
+            content_id, 'Downloading video JSON', fatal=False, expected_status=(404))
 
-        title = media['name'].strip()
+        if media is None:
+            return None
 
-        media_type = media['type']
-        if 'Audio' in media_type:
+        if 'Audio' in media['type']:
             relinker_info = {
                 'formats': [{
-                    'format_id': media.get('formatoAudio'),
+                    'format_id': join_nonempty('https', media.get('formatoAudio'), delim='-'),
                     'url': media['audioUrl'],
                     'ext': media.get('formatoAudio'),
+                    'vcodec': 'none',
+                    'acodec': media.get('formatoAudio'),
                 }]
             }
-        elif 'Video' in media_type:
+        elif 'Video' in media['type']:
             relinker_info = self._extract_relinker_info(media['mediaUri'], content_id)
         else:
             raise ExtractorError('not a media file')
 
-        thumbnails = []
-        for image_type in ('image', 'image_medium', 'image_300'):
-            thumbnail_url = media.get(image_type)
-            if thumbnail_url:
-                thumbnails.append({
-                    'url': compat_urlparse.urljoin(url, thumbnail_url),
-                })
-
-        subtitles = self._extract_subtitles(url, media)
+        thumbnails = self._get_thumbnails_list(
+            {image_type: media.get(image_type) for image_type in (
+                'image', 'image_medium', 'image_300')}, url)
 
         return {
             'id': content_id,
-            'title': title,
-            'description': strip_or_none(media.get('desc') or None),
+            'title': strip_or_none(media.get('name') or media.get('title')),
+            'description': strip_or_none(media.get('desc')) or None,
             'thumbnails': thumbnails,
-            'uploader': strip_or_none(media.get('author') or None),
+            'uploader': strip_or_none(media.get('author')) or None,
             'upload_date': unified_strdate(media.get('date')),
             'duration': parse_duration(media.get('length')),
-            'subtitles': subtitles,
+            'subtitles': self._extract_subtitles(url, media),
             **relinker_info
         }
 
     def _real_extract(self, url):
+        content_info = self._extract_from_content_id(self._match_id(url), url)
+        return content_info or self._legacy_real_extract(url)
+
+    def _legacy_real_extract(self, url):
```

Comment:
I am in favor of deleting this completely unless a usecase is known.

Reply:
for me it's ok to scrap the legacy method.

I just left it there as a fallback solution for maybe some weird webpages that might still exist but that I couldn't find.
But again, I'm 99% sure this is useless.